autoload/arduino.vim

```diff
@@ -418,18 +463,31 @@ function! arduino#GetArduinoDir()
   if exists('g:arduino_dir')
     return g:arduino_dir
   endif
-  let executable = arduino#GetArduinoExecutable()
-  let arduino_cmd = arduino#FindExecutable(executable)
-  let arduino_dir = fnamemodify(arduino_cmd, ':h')
-  if s:OS == 'Darwin'
-    let arduino_dir = fnamemodify(arduino_dir, ':h') . '/Java'
-  endif
-  if !s:FileExists(arduino_dir . '/hardware/arduino/')
-    throw "Could not find arduino directory. Please set g:arduino_dir"
+  if arduino#GetArduinoUserInstallation() == 1
+    let arduino_dir = $HOME . '/.arduino15'
+    if !s:FileExists(arduino_dir)
+      throw "Could not find arduino directory. Please set g:arduino_dir"
+    endif
+  else
+    let executable = arduino#GetArduinoExecutable()
+    let arduino_cmd = arduino#FindExecutable(executable)
+    let arduino_dir = fnamemodify(arduino_cmd, ':h')
+    if s:OS == 'Darwin'
+      let arduino_dir = fnamemodify(arduino_dir, ':h') . '/Java'
+    endif
+    if !s:FileExists(arduino_dir . '/hardware/arduino/')
+      throw "Could not find arduino directory. Please set g:arduino_dir"
+    endif
   endif
   return arduino_dir
 endfunction
 
+function! arduino#GetArduinoUserInstallation()
```

Comment:
I don't think you actually need this function. Since you do the `exists()` check at the top of the file, any time you're calling this function you can instead just read out the value of `g:arduino_user_installation`

Reply:
The user should be able to configure the directory, because it can be configured in the arduino ide as well.
So I think I'll need this check to verify the existence of the directory.

Reply:
Perhaps I'm misunderstanding, but at the top of the file you're doing
```vimL
if !exists('g:arduino_user_installation')
  if s:FileExists(s:ARDUINO_USER_DIR)
    let g:arduino_user_installation = 1
  else
    let g:arduino_user_installation = 0
  endif
endif
```
So by the time this method is called, the only way `exists('g:arduino_user_installation')` will be false is if the user has manually run `unlet g:arduino_user_installation` somewhere. It seems like you could just access `g:arduino_user_installation` directly instead of calling this getter.